lib/remap-functions/src/get_env_var.rs

```diff
@@ -0,0 +1,86 @@
+use remap::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct GetEnvVar;
+
+impl Function for GetEnvVar {
+    fn identifier(&self) -> &'static str {
+        "get_env_var"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "name",
+            accepts: |v| matches!(v, Value::Bytes(_)),
+            required: true,
+        }]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let name = arguments.required("name")?.boxed();
+
+        Ok(Box::new(GetEnvVarFn { name }))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct GetEnvVarFn {
+    name: Box<dyn Expression>,
+}
+
+impl GetEnvVarFn {
+    #[cfg(test)]
+    fn new(name: Box<dyn Expression>) -> Self {
+        Self { name }
+    }
+}
+
+impl Expression for GetEnvVarFn {
+    fn execute(&self, state: &mut state::Program, object: &mut dyn Object) -> Result<Value> {
+        let bytes = self.name.execute(state, object)?.try_bytes()?;
+        let name = String::from_utf8_lossy(&bytes);
+
+        let value = std::env::var(name.as_ref())
+            .map_err(|_| Error::Call(format!("Environment variable {:?} not found.", name)))?;
+        Ok(value.into())
+    }
+
+    fn type_def(&self, state: &state::Compiler) -> TypeDef {
+        self.name
+            .type_def(state)
+            .fallible_unless(value::Kind::Bytes)
```

Comment:
```suggestion
            .into_fallible(true)
```

This function will have to always be fallible, since we don't know if an environment variable exists or not.

Maybe we could check if the variable exists in this function (which runs at boot), and mark it as fallible if it doesn't? I _think_ [there's no way to remove the variable once a process has started](https://unix.stackexchange.com/a/38212/52226), without using hacks. So, I feel we can be safe doing so.

cc @jszwedko.

Reply:
I think that wouldn't cover cases where the env var name is dynamic, for example:

```
value = get_env_var(.some_env_var)
```

Reply:
Ah of course. Yes.